### PR TITLE
[TDF] Fix reading of c-style arrays from jitted transformations and actions

### PR DESCRIPTION
**Pull request by bluehood**
Solves the reading-related part of issue [ROOT-8979](https://sft.its.cern.ch/jira/browse/ROOT-8979).

### CODE REVIEW

**Comment by dpiparo** [comment]
waiting for jenkins for merging...

File: core/foundation/inc/ROOT/TypeTraits.hxx [hunk 1]
```diff
@@ -63,6 +64,11 @@ struct IsContainer {
    static constexpr bool value = Test<Test_t>(nullptr);
 };
 
+template<typename T>
+struct IsContainer<std::array_view<T>> {
+   static constexpr bool value = true;
+};
```
Comment (amadio) [inline core/foundation/inc/ROOT/TypeTraits.hxx]: Why is this here and not in `ROOT/RArrayView.hxx`? I don't understand.

Comment (dpiparo) [reply core/foundation/inc/ROOT/TypeTraits.hxx]: I think because we would like to have the TypeTraits to be aware of the array_view and not the opposite.

Comment (amadio) [reply core/foundation/inc/ROOT/TypeTraits.hxx]: Ah, sorry, somehow I though that this was a TDF header... makes sense.

**Comment by dpiparo** [comment]
Thanks!
